README.md

```diff
@@ -1,11 +1,87 @@
+
+#### Deposits
+Deposits can happen at any time during the pooling period. 
+A participant can post multiple deposits at different times. 
+The contract computes the value of the deposit compared to the total balance it currently holds and eventually uses this "share" - combined with the time of deposit - to calculate the share of MGN that each participation contributed (see below).
+Note, that a deposit only starts contributing to the MGN generation in the next even auction index (since the contract can only invest it in the *depositToken -> secondaryToken* auction)
+
+#### Deposit Withdraws
+Participants can withdraw their deposit **only after the pooling period has ended**.
+Note, that due to price fluctuations it is possible that people withdraw a different amount than they deposited. 
+E.g. if they intially provided 10% of the pool and the price of GNO fell over the time of pooling, the contract will hold less GNO at the end.
+Particpants are only entitled to withdraw their share, which would then be worth less than the initial deposit.
+
+Note, that in order to avoid adjusting all shares after each deposit, we don't store the relative contribution of each participation in %, but instead store an absolute number of *pool shares* (more gas efficient).
+`calculatePoolShares()` contains the logic for computing the number of pool shares a deposit is credited with.
+
+#### Magnolia Withdraws
+On top of their deposit share, participants can also claim their share of MGN tokens at the end.
+We calculate the share by keeping an integral of pool shares that participated in auctions over time.
+For each participation we can then compute their share by computing `lengthOfParticipation * shareOfParticipation / integral`
+
+E.g. assume there were 50 shares (1st deposit 50GNO) in the first round, and 75 shares (2nd deposit 25GNO) in the second round, the integral would be 125 shares.
+The first deposit would be entitled to `2 * 50 / 125 = 80%` of the minted magnolia.
+The second deposit would be entitled to `1 * 25 / 125 = 20%` of the minted magnolia.
+
+#### "Automated" Particpation
+
+We deploy a bot (found under scripts) that will run continuously and check if the pools can participate in the next aution.
+If so, the bot executes the according transaction.
+
+Once the pooling period is over, we call `triggerMGNunlockAndClaimTokens()` manually. 
+This will claim all remaining proceeds from the dutchX back into the pools deposit token balance to allow participants to withdraw their deposits.
+It will also unlock all minted MGN.
+
+Due to the way MGN works, we have to wait 24 hours more, before claiming the unlocked MGN into the pool's balance (by invoking `withdrawUnlockedMagnoliaFromDx()`).
+At that point participants can withdraw their share of MGN.
+
+In principle any party can invoke those calls. 
+There are no permission checks that are relying on our bot invoking these functions.
```

Comment:
This seems a little off? Does this mean there are no permission checks that would prevent the bot from auction participation?

Reply:
Changed it to "There are no permission checks that are relying on our bot invoking these functions."